app/assets/javascripts/analytics/pii.js

```diff
@@ -0,0 +1,64 @@
+;(function (global) {
+  'use strict'
+
+  var GOVUK = global.GOVUK || {}
+  var EMAIL_PATTERN = /[^\s=/?&]+(?:@|%40)[^\s=/?&]+/g
+  var POSTCODE_PATTERN = /[A-PR-UWYZ][A-HJ-Z]?[0-9][0-9A-HJKMNPR-Y]?(?:[\s+]|%20)*[0-9][ABD-HJLNPQ-Z]{2}/gi
+  var DATE_PATTERN = /\d{4}(-?)\d{2}(-?)\d{2}/g
```

Comment:
This is quite specific to one format.
For example:
```
var DATE_PATTERN = /\d{4}(-?)\d{2}(-?)\d{2}/g
var dates = ['2019-01-02', '2019/09/12', '02/02/2019', '02-02-2019', '02.02.2019']
dates.forEach((date) => console.log(date, date.replace(DATE_PATTERN, '[date]')))
```
Only the '2019-01-02' date was found to have a date in it. It might be worth writing some tests to capture more variations of emails, dates and postcodes, to find out those we might be missing.